### ISSUE TITLE
'expected variable-declaration to have a typedef' lint error on for...of loops

### ISSUE THREAD

**Issue by SpenceDiNicolantonio** [palantir/tslint#743]
When using `"typedef": [true, "variable-declaration"]`,`tslint` complains about variables within for...of loop declarations; however typedefs are not allowed in this context.

The following will result in a `expected variable-declaration: 'thing' to have a typedef ` error from `tslint`:
```Typescript
for (let thing of setOfThings) {
    ...
}
``` 
but specifying a typedef on `thing` results in the following `tsc` compilation error:
`error TS2483: The left-hand side of a 'for...of' statement cannot use a type annotation`

The `typedef` rule should ignore variables declared within for...of loop declarations

**Comment by JKillian** [comment]
Thanks for the report! I've filed a PR to get that fixed up
